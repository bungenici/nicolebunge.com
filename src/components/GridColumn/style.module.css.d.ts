/* eslint-disable @typescript-eslint/camelcase */

export const grid__column: string;
export const grid__columnColumnAlignCenter: string;
export const grid__columnColumnAlignEnd: string;
export const grid__columnColumnAlignStart: string;
export const grid__columnColumnSpan1: string;
export const grid__columnColumnSpan10: string;
export const grid__columnColumnSpan11: string;
export const grid__columnColumnSpan12: string;
export const grid__columnColumnSpan2: string;
export const grid__columnColumnSpan3: string;
export const grid__columnColumnSpan4: string;
export const grid__columnColumnSpan5: string;
export const grid__columnColumnSpan6: string;
export const grid__columnColumnSpan7: string;
export const grid__columnColumnSpan8: string;
export const grid__columnColumnSpan9: string;
export const grid__columnColumnStart1: string;
export const grid__columnColumnStart10: string;
export const grid__columnColumnStart11: string;
export const grid__columnColumnStart12: string;
export const grid__columnColumnStart2: string;
export const grid__columnColumnStart3: string;
export const grid__columnColumnStart4: string;
export const grid__columnColumnStart5: string;
export const grid__columnColumnStart6: string;
export const grid__columnColumnStart7: string;
export const grid__columnColumnStart8: string;
export const grid__columnColumnStart9: string;
export const grid__columnRowAlignCenter: string;
export const grid__columnRowAlignEnd: string;
export const grid__columnRowAlignStart: string;
