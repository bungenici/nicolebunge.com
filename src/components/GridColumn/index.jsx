import classNames from 'classnames';
import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import styles from './style.module.css';

function capitalize(string) {
  return `${string.charAt(0).toUpperCase()}${string.slice(1)}`;
}

class GridColumn extends PureComponent {
  static propTypes = {
    children: PropTypes.node,
    columnAlign: PropTypes.string,
    columnSpan: PropTypes.number,
    columnStart: PropTypes.number,
    rowAlign: PropTypes.string,
  };

  static defaultProps = {
    children: null,
    columnAlign: '',
    columnSpan: 1,
    columnStart: 1,
    rowAlign: '',
  };

  render() {
    const { children, columnAlign, columnSpan, columnStart, rowAlign } = this.props;

    return (
      <div
        className={classNames(styles.grid__column, {
          [styles[`grid__columnColumnSpan${columnSpan}`]]: columnSpan,
          [styles[`grid__columnColumnStart${columnStart}`]]: columnStart,
          [styles[`grid__columnColumnAlign${capitalize(columnAlign)}`]]: columnAlign,
          [styles[`grid__columnRowAlign${capitalize(rowAlign)}`]]: rowAlign,
        })}
      >
        {children}
      </div>
    );
  }
}

export default GridColumn;
