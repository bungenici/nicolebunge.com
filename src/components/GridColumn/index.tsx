import classNames from 'classnames';
import React from 'react';
import styles from './style.module.css';

function capitalize(string = ''): string {
  return `${string.charAt(0).toUpperCase()}${string.slice(1)}`;
}

type GridColumnSpanClass =
  | 'grid__columnColumnSpan1'
  | 'grid__columnColumnSpan2'
  | 'grid__columnColumnSpan3'
  | 'grid__columnColumnSpan4'
  | 'grid__columnColumnSpan5'
  | 'grid__columnColumnSpan6'
  | 'grid__columnColumnSpan7'
  | 'grid__columnColumnSpan8'
  | 'grid__columnColumnSpan9'
  | 'grid__columnColumnSpan10'
  | 'grid__columnColumnSpan11'
  | 'grid__columnColumnSpan12';

type GridColumnStartClass =
  | 'grid__columnColumnStart1'
  | 'grid__columnColumnStart2'
  | 'grid__columnColumnStart3'
  | 'grid__columnColumnStart4'
  | 'grid__columnColumnStart5'
  | 'grid__columnColumnStart6'
  | 'grid__columnColumnStart7'
  | 'grid__columnColumnStart8'
  | 'grid__columnColumnStart9'
  | 'grid__columnColumnStart10'
  | 'grid__columnColumnStart11'
  | 'grid__columnColumnStart12';

type GridColumnAlignClass =
  | 'grid__columnColumnAlignCenter'
  | 'grid__columnColumnAlignEnd'
  | 'grid__columnColumnAlignStart';

type GridRowAlignClass =
  | 'grid__columnRowAlignCenter'
  | 'grid__columnRowAlignEnd'
  | 'grid__columnRowAlignStart';

export interface GridColumnProps {
  className?: string;
  columnAlign?: string;
  columnSpan?: number;
  columnStart?: number;
  rowAlign?: string;
}

const GridColumn: React.FC<GridColumnProps> = (props) => {
  const {
    children,
    className,
    columnAlign,
    columnSpan,
    columnStart,
    rowAlign,
    ...otherProps
  } = props;

  return (
    <div
      className={classNames(className, styles.grid__column, {
        [styles[`grid__columnColumnSpan${columnSpan}` as GridColumnSpanClass]]: columnSpan,
        [styles[`grid__columnColumnStart${columnStart}` as GridColumnStartClass]]: columnStart,
        [styles[
          `grid__columnColumnAlign${capitalize(columnAlign)}` as GridColumnAlignClass
        ]]: columnAlign,
        [styles[`grid__columnRowAlign${capitalize(rowAlign)}` as GridRowAlignClass]]: rowAlign,
      })}
      {...otherProps}
    >
      {children}
    </div>
  );
};

GridColumn.defaultProps = {
  columnSpan: 1,
  columnStart: 1,
};

export default GridColumn;
