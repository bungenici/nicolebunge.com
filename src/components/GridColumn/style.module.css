.grid__column > :first-child {
  margin-top: 0;
}

.grid__column > :last-child {
  margin-bottom: 0;
}

.grid__column--column-align-start {
  justify-self: flex-start;
}

.grid__column--column-align-center {
  justify-self: center;
}

.grid__column--column-align-end {
  justify-self: flex-end;
}

.grid__column--column-span-1 {
  grid-column-end: span 1;
}

.grid__column--column-span-2 {
  grid-column-end: span 2;
}

.grid__column--column-span-3 {
  grid-column-end: span 3;
}

.grid__column--column-span-4 {
  grid-column-end: span 4;
}

.grid__column--column-span-5 {
  grid-column-end: span 5;
}

.grid__column--column-span-6 {
  grid-column-end: span 6;
}

.grid__column--column-span-7 {
  grid-column-end: span 7;
}

.grid__column--column-span-8 {
  grid-column-end: span 8;
}

.grid__column--column-span-9 {
  grid-column-end: span 9;
}

.grid__column--column-span-10 {
  grid-column-end: span 10;
}

.grid__column--column-span-11 {
  grid-column-end: span 11;
}

.grid__column--column-span-12 {
  grid-column-end: span 12;
}

.grid__column--column-start-1 {
  grid-column-start: 1;
}

.grid__column--column-start-2 {
  grid-column-start: 2;
}

.grid__column--column-start-3 {
  grid-column-start: 3;
}

.grid__column--column-start-4 {
  grid-column-start: 4;
}

.grid__column--column-start-5 {
  grid-column-start: 5;
}

.grid__column--column-start-6 {
  grid-column-start: 6;
}

.grid__column--column-start-7 {
  grid-column-start: 7;
}

.grid__column--column-start-8 {
  grid-column-start: 8;
}

.grid__column--column-start-9 {
  grid-column-start: 9;
}

.grid__column--column-start-10 {
  grid-column-start: 10;
}

.grid__column--column-start-11 {
  grid-column-start: 11;
}

.grid__column--column-start-12 {
  grid-column-start: 12;
}

.grid__column--row-align-start {
  align-self: flex-start;
}

.grid__column--row-align-center {
  align-self: center;
}

.grid__column--row-align-end {
  align-self: flex-end;
}
