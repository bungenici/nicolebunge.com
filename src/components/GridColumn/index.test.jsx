import React from 'react';
import renderer from 'react-test-renderer';
import GridColumn from '.';

describe('GridColumn component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<GridColumn />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
