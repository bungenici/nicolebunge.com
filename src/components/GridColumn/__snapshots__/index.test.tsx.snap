// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridColumn component renders correctly 1`] = `
<div
  className="grid__column grid__columnColumnSpan1 grid__columnColumnStart1"
/>
`;
